### PR TITLE
Feature/40195 seed data for new team planner module

### PR DESCRIPTION
**Pull request by ulferts**
Work packages "Project plan"-View:

- ~Days Zoom level~ respecified
- [x] Move start date of the work packages (2, 3 and the new ones) to the Monday of the creation date of the instance
- [x] Move project start work packages (closed) to the Sunday before
- [x] Save as Project plan (for public and favourite views)
- [x] Switch tasks 4 and 5 so that Invite attendees to conference will be displayed above Setup conference website
- [x] Add three work packages as child element of Set date and location of conference:
- ~Adapt filter so that the closed project milestone is visible~

image

Module team planner

- [x] Activate the module in the Demo project
- [x] Create a board: "Team planner" and set it to Favored and Public.
- [x] Add the user who created the trial to the board

image

BIM edition:

- [x] Activate the team planner module in all 4 demo projects and seed a board.
- [x] In Work Packages module starred the "Project plan" query. 

https://community.openproject.org/wp/40195

### CODE REVIEW

File: app/seeders/demo_data/project_seeder.rb [hunk 1]
```diff
@@ -134,7 +134,7 @@ def set_project_status(project, key)
 
     def set_members(project)
       role = Role.find_by(name: translate_with_base_url(:default_role_project_admin))
-      user = User.admin.first
+      user = User.user.admin.first
```
Comment (wielinde) [inline app/seeders/demo_data/project_seeder.rb:137]: Why is that change necessary? Shouldn't single table inheritance just filter exactly that?

File: app/seeders/demo_data/query_builder.rb [hunk 2]
```diff
@@ -50,10 +50,11 @@ def valid?
     def base_attributes
       {
         name: config[:name],
-        user: User.admin.first,
-        public: config[:public] != false,
-        show_hierarchies: config[:hierarchy] == true,
-        timeline_visible: config[:timeline] == true
+        user: User.admin.user.first,
```
Comment (wielinde) [inline app/seeders/demo_data/query_builder.rb:53]: again, I don't understand why we need the `.user` scope here

File: app/seeders/demo_data/work_package_seeder.rb [hunk 3]
```diff
@@ -36,7 +36,7 @@ class WorkPackageSeeder < Seeder
     def initialize(project, key)
       self.project = project
       self.key = key
-      self.user = User.admin.first
+      self.user = User.user.admin.first
```
Comment (wielinde) [inline app/seeders/demo_data/work_package_seeder.rb:39]: again

File: app/seeders/demo_data/project_seeder.rb (hunk 1 above)
Comment (ulferts) [reply app/seeders/demo_data/project_seeder.rb:137]:
```
pry(main)> User.admin.to_sql
=> "SELECT \"users\".* FROM \"users\" WHERE \"users\".\"type\" IN ('User', 'SystemUser') AND \"users\".\"admin\" = TRUE"
[6] pry(main)> User.user.admin.to_sql
=> "SELECT \"users\".* FROM \"users\" WHERE \"users\".\"type\" IN ('User', 'SystemUser') AND \"users\".\"type\" = 'User' AND \"users\".\"admin\" = TRUE"
```

In my test runs, the SystemUser was assigned sometimes, based on the whim of the database as to what is first.

File: app/seeders/demo_data/work_package_seeder.rb [hunk 4]
```diff
@@ -207,12 +207,13 @@ def create_relation(to:, from:, type:)
...
-      monday = Date.today.monday
-      days_ahead > 0 ? monday + days_ahead : monday
+      Time.zone.today.monday + (days_ahead || 0).days
     end
 
+    # Returns the due date based on the starting date and the duration
+    # but ensures that the due date cannot be before the start date.
     def calculate_due_date(date, duration)
-      duration && duration > 1 ? date + duration : date
+      [date + ((duration || 0) - 1).days, date].max
```
Comment (wielinde) [inline app/seeders/demo_data/work_package_seeder.rb:216]: This one here looks much better than the older line. However, I believe that they produce different results as we now subtract one day from the duration. Is that a problem?

File: app/seeders/demo_data/project_seeder.rb (hunk 1 above)
Comment (wielinde) [reply app/seeders/demo_data/project_seeder.rb:137]: Gotcha.

File: app/seeders/demo_data/work_package_seeder.rb (hunk 4 above)
Comment (ulferts) [reply app/seeders/demo_data/work_package_seeder.rb:216]: The manually groomed seed data has been adapted and looks fine AFAIK. But for BIM, the seed pulling might need to be adapted.

**Comment by ulferts** [comment]
@wielinde all open questions for the vanilla edition have been clarified. If it is all right with you, I'd like to merge this in even though BIM might be affected so that work on the onboarding tour can begin.

Please remove the draft status if you agree (and ideally review it).

**Review by wielinde — APPROVED** [review]
LGTM.

**Comment by wielinde** [comment]
@ulferts I added Team Planner boards to the BIM demo data and did not touch anything else. BTW: Thumbs up for the solution to set the assignee to the first human admin. It looks so much better know. Let's see how many admins will complain about updates they receive as other users update those WPs :-)
